cpu/ppc64_cpu_test.py

```diff
@@ -111,56 +112,68 @@ def smt(self):
         """
         Tests the SMT in ppc64_cpu command.
         """
-        command1 = "ppc64_cpu --smt | awk -F'=' '{print $NF}' | awk \
-                '{print $NF}'"
-        command2 = "echo %s" % self.value
-        self.equality_check("SMT", command1, command2)
+        op1 = process.system_output(
+            "ppc64_cpu --smt", shell=True).strip().split("=")[-1].split()[-1]
+        self.equality_check("SMT", op1, self.value)
 
     def core(self):
         """
         Tests the core in ppc64_cpu command.
         """
-        command1 = "ppc64_cpu --cores-present | awk '{print $NF}'"
-        command2 = "expr $(grep -w processor /proc/cpuinfo | wc -l) / %d" \
-            % self.key
-        self.equality_check("Core", command1, command2)
+        op1 = process.system_output(
+            "ppc64_cpu --cores-present", shell=True).strip().split()[-1]
+        count = 0
+        for line in genio.read_file("/proc/cpuinfo").strip().splitlines():
+            if line.startswith('processor'):
+                count += 1
+        op2 = count / int(self.key)
+        self.equality_check("Core", op1, op2)
```

Comment:
this will not be true for cases where SMT !=off,
only when SMT=off, threads = 1, no. of processors = cores otherwise processors > cores_present.

```
>>> from avocado.utils import genio
>>> count=0
>>> for line in genio.read_file("/proc/cpuinfo").strip().splitlines():
...     if line.startswith('processor'):
...             count +=1
... 
>>> count
160
>>> 
[root@9 ~]# ppc64_cpu --cores-present
Number of cores present = 40
[root@9 ~]# ppc64_cpu --smt
SMT=4
[root@9 ~```